pkg/api/controller_test.go

```diff
@@ -3083,3 +3083,90 @@ func TestController_GetPhysicalAddress_(t *testing.T) {
+		}
+		if resp.JSON400 == nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse expected BadRequest: %+v", resp)
+		}
+	})
+
+	t.Run("repository_not_exists", func(t *testing.T) {
+		repo := testUniqueRepoName()
+		resp, err := clt.PrepareGarbageCollectionUncommittedWithResponse(ctx, repo, api.PrepareGarbageCollectionUncommittedJSONRequestBody{})
+		if err != nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse failed: %s", err)
+		}
+		if resp.JSON404 == nil {
+			t.Fatalf("PrepareGarbageCollectionUncommittedWithResponse expected NotFound: %+v", resp)
+		}
+	})
+
+	t.Run("uncommitted", func(t *testing.T) {
+		repo := testUniqueRepoName()
+		_, err := deps.catalog.CreateRepository(ctx, repo, onBlock(deps, repo), "main")
+		testutil.Must(t, err)
+		const items = 3
+		for i := 0; i < items; i++ {
+			path := fmt.Sprintf("uncommitted/obj%d", i)
+			uploadResp, err := uploadObjectHelper(t, ctx, clt, path, strings.NewReader(path), repo, "main")
+			verifyResponseOK(t, uploadResp, err)
+		}
+		var continuationToken *string
+		for {
+			resp, err := clt.PrepareGarbageCollectionUncommittedWithResponse(ctx, repo, api.PrepareGarbageCollectionUncommittedJSONRequestBody{
+				ContinuationToken: continuationToken,
+			})
+			verifyResponseOK(t, resp, err)
+			if resp.JSON201.RunId == "" {
+				t.Errorf("PrepareGarbageCollectionUncommittedWithResponse empty RunID, value expected")
+			}
+			if resp.JSON201.Location == "" {
+				t.Errorf("PrepareGarbageCollectionUncommittedWithResponse empty Location, value expected")
+			}
+			token := api.StringValue(resp.JSON201.ContinuationToken)
+			if token == "" {
+				break
+			}
+		}
+	})
+
+	t.Run("committed", func(t *testing.T) {
+		repo := testUniqueRepoName()
+		_, err := deps.catalog.CreateRepository(ctx, repo, onBlock(deps, repo), "main")
+		testutil.Must(t, err)
+		const items = 3
+		for i := 0; i < items; i++ {
+			path := fmt.Sprintf("committed/obj%d", i)
+			uploadResp, err := uploadObjectHelper(t, ctx, clt, path, strings.NewReader(path), repo, "main")
+			verifyResponseOK(t, uploadResp, err)
+		}
+		if _, err := deps.catalog.Commit(ctx, repo, "main", "committed objects", "some_user", nil, nil, nil); err != nil {
+			t.Fatalf("failed to commit objects: %s", err)
+		}
+		var continuationToken *string
+		for {
+			resp, err := clt.PrepareGarbageCollectionUncommittedWithResponse(ctx, repo, api.PrepareGarbageCollectionUncommittedJSONRequestBody{
+				ContinuationToken: continuationToken,
+			})
+			verifyResponseOK(t, resp, err)
+			if resp.JSON201.RunId == "" {
+				t.Errorf("PrepareGarbageCollectionUncommittedWithResponse empty RunID, value expected")
+			}
+			if resp.JSON201.Location != "" {
+				t.Errorf("PrepareGarbageCollectionUncommittedWithResponse Location=%s, expected empty", resp.JSON201.Location)
+			}
+			token := api.StringValue(resp.JSON201.ContinuationToken)
+			if token == "" {
+				break
+			}
+		}
+	})
+}
```

Comment:
Maybe add a check the ensures we get at least 1 cont token?

Reply:
Will depend on the catalog test for this one as generating 20MB using the API during tests will just slow the system down.
We will need to have this parameter configurable and test the token.